CavalcanteLucas/labdale
# Feature/OL43-add_todo

**Link to task:** 
https://labcodes.atlassian.net/browse/OL-43

**How to test:**
- With a user logged in having one or more TodoLists, click on the "+" (plus) button in the TodoListDetail page, check if modal with form pops up;
- Fill the form and check if Todo is inserted at the bottom of the listing of Todos.

**Description of your solution:**
This PR implements the API endpoint for enabling CREATE requests of a Todo instance. This PR also creates the interface to action that fetches the CREATE endpoint and handle state updates / redirect action for the REQUEST / SUCCESS / FAILURE cases. Tests ensure that creating Todos require Token authorization, and that it is a private method among different users. A "+" (plus) button in the botton of TodoListDetail page represents the "add Todo" button. When this button is selected, a modal with a form is open. After proper filling, an instance of Todo is created and included in the current page.
